torch_geometric/nn/models/label_usage.py

```diff
@@ -0,0 +1,92 @@
+import torch
+import torch.nn.functional as F
+
+from torch_geometric.typing import OptTensor, Adj
+
+class LabelUsage(torch.nn.Module):
+    r"""The label usage operator for semi-supervised node classification,
+    as introduced in `"Bag of Tricks for Node Classification"
+    <https://arxiv.org/abs/2103.13355>`_ paper.
+
+    .. note::
+
+        When using the :class:`LabelUsage`, adjust the model's input dimension
+        accordingly to include both features and classes.
+
+    Args:
+        base_model: An instance of the model that will do the 
+            inner forward pass.
+        num_classes (int): Number of classes in dataset
+        split_ratio (float): Proportion of true labels to use as features 
+            during training (default: :obj:'0.5')
+        num_recycling_iterations (int): Number of iterations for the
+            label reuse procedure to cycle predicted soft labels 
+            (default: :obj:'100')
+        return_tuple (bool): If true, returns (output, train_label_idx, 
+            train_pred_idx) otherwise returns prediction output 
+            (default :obj:'False')
+    """
+
+    def __init__(
+        self,
+        base_model: torch.nn.Module,
+        num_classes: int,
+        split_ratio: float = 0.5,
+        num_recycling_iterations: int = 100,
```

Comment:
100 seems high, that would mean ~100x forward passes during training.  

I don't think paper says how many they used, and the only value besides zero I'm seeing in their repo is 1.  

1 does seem kind of low, I'd be interested to see how much extra forward passes slow things down on these datasets.  I'd like try values like [0,1,2,3,5,10,100], but we probably don't have time for all of those.  

Let's maybe change the default to 1 for now and see how much it slows things down on `benchmarks/citation` before trying a couple other values if we have time?